contrib/mesos/pkg/scheduler/service/service.go

```diff
@@ -389,7 +391,10 @@ func (s *SchedulerServer) prepareExecutorInfo(hks hyperkube.Interface) (*mesos.E
 	} else if !hks.FindServer(hyperkube.CommandMinion) {
 		return nil, fmt.Errorf("either run this scheduler via km or else --executor-path is required")
 	} else {
-		if strings.Index(s.kmPath, "://") > 0 {
+		if s.kmLocalPath != "" { // The default value of kmLocalPath is "".
+			// TODO(k82): Check whether `kmLocalPath` start with `file://` or `/`.
```

Comment:
The idea being that people wanting to use a preinstalled `km` on the agent would always have to specify a URI using the `file://` form